src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -715,15 +715,10 @@ private async ValueTask<int> EnsureFullTlsFrameAsync<TIOAdapter>(CancellationTok
                 return frameSize;
             }
 
-            if (frameSize < int.MaxValue)
-            {
-                _buffer.EnsureAvailableSpace(frameSize - _buffer.EncryptedLength);
-            }
-
             while (_buffer.EncryptedLength < frameSize)
             {
-                // there should be space left to read into
-                Debug.Assert(_buffer.AvailableLength > 0, "_buffer.AvailableBytes > 0");
+                // make sure we have space to read into
+                _buffer.EnsureAvailableSpace(Math.Min(frameSize, _buffer.Capacity) - _buffer.EncryptedLength);
```

Comment:
Is there risk of growing too much if framers are keep coming. I know this is not normal case but as far as I understand the old code, we would make space for a frame, maybe more if needed. 
We had `EnsureAvailableSpace` on line 246 so I'm wondering why that was not sufficient.

Reply:
yes, there is, I should've waited with the fix till the morning :D

> We had EnsureAvailableSpace on line 246 so I'm wondering why that was not sufficient.

Not sure which one you are talking about, the closes one is in 214 and 287 and both are outside of the while loop which is receiving TLS frames